homeassistant/components/binary_sensor/tcp.py

```diff
@@ -7,21 +7,17 @@
 import logging
 
 from homeassistant.components.binary_sensor import BinarySensorDevice
-from homeassistant.components.sensor.tcp import Sensor, CONF_VALUE_ON
-
+from homeassistant.components.sensor.tcp import TcpSensor, CONF_VALUE_ON
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Create the binary sensor."""
-    if not BinarySensor.validate_config(config):
```

Comment:
You are removing the call to validate the config but did not add a PLATFORM_SCHEMA.